marketplace-builder/src/builder.rs

```diff
@@ -563,40 +606,55 @@ mod test {
         let unregistered_transaction =
             Transaction::new(UNREGISTERED_NAMESPACE.into(), vec![1, 1, 1, 2]);
         let transactions = vec![registered_transaction.clone(), unregistered_transaction];
-        let mut subscribed_events = submit_transactions(transactions, urls).await;
-        task::sleep(std::time::Duration::from_millis(1000)).await;
-
-        // Verify the bundle.
-        let start = Instant::now();
-        loop {
-            if start.elapsed() > Duration::from_secs(10) {
-                panic!("Didn't get a quorum proposal in 10 seconds");
+        if public_mempool {
+            let server = &network.server;
+            for transaction in transactions {
+                let _ = server.submit_transaction(transaction).await;
```

Comment:
Shouldn't we call unwrap on the result? If we fail to submit transactions the test will fail later anyways, and this way the reason will be clearer